src/main/java/hudson/plugins/jira/JiraBuildAction.java

```diff
@@ -27,6 +27,11 @@ public JiraBuildAction(@Nonnull Set<JiraIssue> issues) {
     this.issues = new HashSet<>(issues);
   }
 
+  public JiraBuildAction(@Nonnull Run<?, ?> owner, @Nonnull Set<JiraIssue> issues) {
+    this.owner = owner;
```

Comment:
IIUC what @jglick said about `RunAction2` (quite unfriendly name btw 😆 ), since this is now set by `onLoad`, this line should be removed?

Reply:
Checked the implementation of `Run#addAction(hudson.model.Action)`, I can confirm the `owner` is set by `jenkins.model.RunAction2#onAttached(hudson.model.Run)`. I'm OK to remove it.

Reply:
Yes, so long as the owner is not used by some method called by the constructor (recall such a situation in the `junit` plugin), you can use `onAttached`.

`RunAction` was the original marker interface but people were storing a non`transient` owner and it was causing mayhem. Jenkins convention is to add a numeric suffix to interfaces that supersede older ones.

Reply:
Thanks for the heads up Jesse. I'm just against any `<identifier>2` naming in any code as for _just me_ it's not [transparent](https://softwareengineering.stackexchange.com/questions/138948/how-do-you-evolve-version-an-interface) and promotes bad practice in general - but that's just me :)